enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,214 @@
+---
+title: declarative-index-config
+authors:
+  - "@anik120"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-07-10
+last-updated: 2020-07-10
+status: implementable  
+---
+
+# Declarative Index Config
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions [optional]
+
+## Summary
+
+This enhancement proposes the use of [Cue](https://cuelang.org/docs/about/) to provide a way to declaratively define an index so that operator authors and pipeline developers can validate the content of an index in a standard way that's also consumable by continuous integration tools. 
```

Comment:
IMO, this enhancement should be more about defining a canonical data format -- as JSON -- than it should be about the meta-language we define its schema in; Cue could then just be a recommended tool to generate indexes if we wanted.